src/datasources/asset/AssetDataSource.ts

```diff
@@ -98,6 +99,22 @@ export class AssetDataSource extends DataSourceBase<AssetMetadataQuery> {
     }
   }
 
+  async metricFindQuery({ minionIds, workspace }: AssetQuery): Promise<MetricFindValue[]> {
+    const conditions = [];
+    if (minionIds?.length) {
```

Comment:
Here we'd probably want to implement a query builder using the builder pattern to reduce complexity